src/wallet/wallet2.cpp

```diff
@@ -430,17 +430,16 @@ std::unique_ptr<tools::wallet2> make_basic(const boost::program_options::variabl
   }
 
   boost::optional<bool> trusted_daemon;
+  THROW_WALLET_EXCEPTION_IF(!command_line::is_arg_defaulted(vm, opts.trusted_daemon) && !command_line::is_arg_defaulted(vm, opts.untrusted_daemon),
```

Comment:
With this simplification, boost::optional<bool> isn't needed.

`bool trusted_daemon = false`

And remove the 

`trusted_daemon = false` assignment below and the `*trusted` dereference in `wallet->init`